### PR TITLE
[PoC] [admin] sonata custom page

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| ...
|New feature| Yes/No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes/No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes/No

----
:globe_with_meridians: Live Preview:
  - https://mg-sonata-custom-page.odin.shopsys.cloud
  - https://cz.mg-sonata-custom-page.odin.shopsys.cloud

### CODE REVIEW

File: packages/administration/src/Controller/DashboardController.php [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+     * @return \Shopsys\FrameworkBundle\Model\Administrator\Administrator
+     */
+    protected function getCurrentAdministrator(): Administrator
+    {
+        /** @var \Shopsys\FrameworkBundle\Model\Administrator\Administrator|null $administrator */
+        $administrator = $this->getUser();
+
+        if ($administrator === null) {
+            throw $this->createAccessDeniedException('No administrator is logged in.');
+        }
```
Comment (TomasLudvik) [inline packages/administration/src/Controller/DashboardController.php:153]: You should be able to do this check in security.yaml, right?

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

**Comment by TomasLudvik** [comment]
I am missing one thing - custom JavaScript for making it sortable. No need to implement saving and other stuff, just make it sortable :)

File: packages/administration/src/Controller/DashboardController.php (hunk 1 above)
Comment (grossmannmartin) [reply packages/administration/src/Controller/DashboardController.php:153]: this code is copy-paste from the current admin. Basically to ensure the admin is not null as controller's `$this->getUser();` may return null.

**Comment by grossmannmartin** [comment]
> I am missing one thing - custom JavaScript for making it sortable. No need to implement saving and other stuff, just make it sortable :)

I see. Custom javascript is now prototyped by @vitek-rostislav along with the advanced form fields. 
Is it enough that we will know the custom JS may be used, so it will be usable even here?

**Comment by vitek-rostislav** [comment]
> I see. Custom javascript is now prototyped by @vitek-rostislav along with the advanced form fields.
Is it enough that we will know the custom JS may be used, so it will be usable even here?

I managed to link our current javascript files in this commit - https://github.com/shopsys/shopsys/pull/3188/commits/09057fe3b7e26f83432951c4d04c5e04afa1ca23 - it is achieved by overriding the base template and including the webpack encore functions (`encore_entry_script_tags` and `encore_entry_link_tags`).

**Review by TomasLudvik — APPROVED** [review]

**Pull request closed without merging**